packages/esm-form-entry-app/src/app/form-schema/form-schema.service.ts

```diff
@@ -193,53 +190,36 @@ export class FormSchemaService {
     return formUuids;
   }
 
-  private traverseForUnlabeledConcepts(o, type?) {
-    let concepts = [];
-    if (o.children) {
-      if (o.children instanceof Array) {
-        const returned = this.traverseRepeatingGroupForUnlabeledConcepts(o.children);
-        return returned;
-      }
-      if (o.children instanceof Object) {
-        for (const key in o.children) {
-          if (o.children.hasOwnProperty(key)) {
-            const question = o.children[key].question;
-            switch (question.renderingType) {
-              case 'page':
-              case 'section':
-              case 'group':
-                const childrenConcepts = this.traverseForUnlabeledConcepts(o.children[key]);
-                concepts = concepts.concat(childrenConcepts);
-                break;
-              case 'repeating':
-                const repeatingConcepts = this.traverseRepeatingGroupForUnlabeledConcepts(o.children[key].children);
-                concepts = concepts.concat(repeatingConcepts);
-                break;
-              default:
-                if (!question.label && question.extras.questionOptions?.concept) {
-                  concepts.push(question.extras.questionOptions.concept);
-                }
-                if (question.extras.questionOptions.answers) {
-                  question.extras.questionOptions.answers.forEach((answer) => {
-                    if (!answer.label) {
-                      concepts.push(answer.concept);
-                    }
-                  });
-                }
-                break;
-            }
+  public static getUnlabeledConceptUuidsFromSchema(form: FormSchema): Array<string> {
```

Comment:
That sounds good, I will change it. Thanks!

Reply:
> but there is a plan underway to create an endpoint which would allow us to refer to concepts by something other than UUIDs, specifically concept maps

@ibacher doesn't the [/concept/CIEL:1065](https://demo.openmrs.org/openmrs/ws/rest/v1/concept/) REST API already support this?

Reply:
@dkayiwa Good point! Really it's less dependent on an endpoint and more dependent on refactoring the frontend code to be able to use those references correctly.